spec/unit/config_files_inspector_spec.rb

```diff
@@ -96,231 +119,343 @@
-      system = double
-      expect_requirements(system)
-      expect_rpm_qa(system,rpm_qa_output_test2)
-      expect(system).to receive(:run_command).with(
-        *base_cmdline, "xml-commons-1.3.04",
-        :stdout => :capture
-      ).and_return("")
-
-      inspector = ConfigFilesInspector.new(system, description)
-      inspector.inspect(filter)
-
-      expected = ConfigFilesScope.new(
-        extracted: false,
-        files: ConfigFileList.new()
-      )
-      expect(description["config_files"]).to eq(expected)
-    end
+    describe "#config_file_changes" do
+      it "returns a list of changed config files" do
 
-    it "raise an error when requirements are not fulfilled" do
-      system = double
-      expect(system).to receive(:check_requirement).with(
-        "rpm", "--version"
-      ).and_raise(Machinery::Errors::MissingRequirement)
+        apache_config_1 = ConfigFile.new(
+            name: "/etc/apache2/default-server.conf",
+            package_name: "apache2",
+            package_version: "2.4.6",
+            status: "changed",
+            changes: ["size", "md5", "time"]
+        )
 
-      inspector = ConfigFilesInspector.new(system, description)
-      expect{inspector.inspect(filter)}.to raise_error(
-        Machinery::Errors::MissingRequirement)
-    end
+        apache_config_2 = ConfigFile.new(
+            name: "/etc/apache2/listen.conf",
+            package_name: "apache2",
+            package_version: "2.4.6",
+            status: "changed",
+            changes: ["md5", "time"]
+        )
 
-    it "extracts changed configuration files" do
-      system = double
-      expect_inspect_configfiles(system, true)
+        apache_config_3 = ConfigFile.new(
+            name: "/etc/sysconfig/SuSEfirewall2.d/services/apache2",
+            package_name: "apache2",
+            package_version: "2.4.6",
+            status: "changed",
+            changes: ["size", "md5", "time"]
+        )
 
-      inspector = ConfigFilesInspector.new(system, description)
-      inspector.inspect(filter, extract_changed_config_files: true)
-      expect(inspector.summary).to include("Extracted 6 changed configuration files")
-      cfdir = File.join(store.description_path(name), "config_files")
-      expect(File.stat(cfdir).mode & 0777).to eq(0700)
-    end
+        apache_config_4 = ConfigFile.new(
+            name: "/usr/share/info/dir",
+            package_name: "apache2",
+            package_version: "2.4.6",
+            status: "changed",
+            changes: ["deleted"]
+        )
+        apache_config_5 = ConfigFile.new(
+            name: "/usr/share/man/man1/time.1.gz",
+            package_name: "apache2",
+            package_version: "2.4.6",
+            status: "changed",
+            changes: ["replaced"]
+        )
+
+        expected_apache_config_file_data = apache_config_1, apache_config_2, apache_config_3,
+					apache_config_4, apache_config_5
```

Comment:
Align the elements of an array literal if they span more than one line.<br>Tab detected.